src/Core/View.js

```diff
@@ -0,0 +1,185 @@
+import MainLoop from './MainLoop';
+import c3DEngine from '../Renderer/c3DEngine';
+import { STRATEGY_MIN_NETWORK_TRAFFIC } from './Layer/LayerUpdateStrategy';
+import Scheduler from './Scheduler/Scheduler';
+import WMTS_Provider from './Scheduler/Providers/WMTS_Provider';
+import WMS_Provider from './Scheduler/Providers/WMS_Provider';
+import TileProvider from './Scheduler/Providers/TileProvider';
+import Debug from '../../utils/debug/Debug';
+
+/**
+ * Constructs an Itowns Scene instance
+ *
+ * @param {string} crs - The default CRS of Three.js coordinates. Should be a cartesian CRS.
+ * @param {Coordinates} positionCamera - The initial position of the camera
+ * @param {DOMElement} viewerDiv - Where to instanciate the Three.js scene in the DOM
+ * @param {boolean} debugMode - activate debug mode
+ * @param {boolean} glDebug - debug gl code
+ * @constructor
+ */
+ /* TODO:
+ * - remove debug boolean, replace by if __DEBUG__ and checkboxes in debug UI
+ * - Scene (and subobjects) should be instanciable several times.
+ */
+function View(crs, viewerDiv, mainLoop) {
+    this.referenceCrs = crs;
+
+    this.mainLoop = mainLoop || new MainLoop(new Scheduler(), new c3DEngine(viewerDiv));
+
+    this.camera = new Camera(
+        viewerDiv.clientWidth,
+        viewerDiv.clientHeight);
+
+    this._layers = [];
+
+    if (__DEBUG__) {
+        this._debug = new Debug(this, viewerDiv);
+    }
+
+    this.viewerDiv = viewerDiv;
+    window.addEventListener('resize', () => {
+        this.mainLoop.gfxEngine.onWindowResize();
+        this.camera.resize(this.viewerDiv.clientWidth, this.viewerDiv.clientHeight);
+        this.camera.update();
+        this.notifyChange(0, true);
+    }, false);
+
+    this.onAfterRender = () => {};
+}
+
+function _preprocessLayer(view, layer, provider) {
+    if (!layer.updateStrategy) {
+        layer.updateStrategy = {
+            type: STRATEGY_MIN_NETWORK_TRAFFIC,
+        };
+    }
+
+    if (provider) {
+        if (provider.tileInsideLimit) {
+            layer.tileInsideLimit = provider.tileInsideLimit.bind(provider);
+        }
+
+        if (provider.tileTextureCount) {
+            layer.tileTextureCount = provider.tileTextureCount.bind(provider);
+        }
+
+        if (provider.preprocessDataLayer) {
+            provider.preprocessDataLayer(layer);
+        }
+    }
+
+    // probably not the best place to do this
+    if (provider instanceof TileProvider || !provider) {
+        // const threejsLayer = view.mainLoop.gfxEngine.getUniqueThreejsLayer();
+        // layer.threejsLayer = threejsLayer;
+        // enable by default
+        // TODO camera.camera3D.layers.enable(threejsLayer);
+    } else {
+        layer.frozen = layer.frozen || false;
+        layer.visible = layer.visible == undefined ? true : layer.visible;
+        layer.opacity = layer.opacity == undefined ? 1.0 : layer.opacity;
+        layer.sequence = 0;
+    }
+}
+
+View.prototype.addLayer = function addLayer(layer, parentLayer) {
```

Comment:
Yeah... mainly because the main usage is to add layers.
We'll probably want to add a `removeLayer` method layer though.